R/Covid19DataHub.R

```diff
@@ -123,11 +124,20 @@ Covid19DataHub <- R6::R6Class("Covid19DataHub",
           recovered_new = as.numeric(.data$recovered_new),
           tested_new = as.numeric(.data$tested_new),
           hosp_new = as.numeric(.data$hosp_new)
-        ) %>%
+        )
+
+      # remove levels not requested
+      all_levels <- paste0("^level_", self$supported_levels, "_*")
```

Comment:
This will drop level 1 regions when we have level 2 data which we don't want? I think we want to just drop levels > than current level?

Reply:
No this works:

`all_levels`  makes a list of `c("^level_1_*", "^level_2_*", "^level_3_*")`
`keep_levels` gets the level regex up to that of interest using `seq_len`: `c("^level_1_*", "^level_2_*")` using `self$level`
`setdiff(all_levels, keep_levels)`  extracts the difference between the two lists and so gives the one not needed (`"^level_3_*"`) and only this one is removed in the grep.

```
a <- Covid19DataHub$new(level = "2", verbose = FALSE)
a$download()
a$clean()
 "level_1_region" %in% colnames(a$data$clean)                                                                                                                                                                           
[1] TRUE
"level_2_region" %in% colnames(a$data$clean)                                                                                                                                                                           
[1] TRUE
"level_3_region" %in% colnames(a$data$clean)                                                                                                                                                                           
[1] FALSE
```

Reply:
ah well that is very clever then!